src/add-archival-events/src/db.ts

```diff
@@ -0,0 +1,163 @@
+import groupBy from "lodash.groupby"
+import type { QueryResult } from "pg"
+import { Client } from "pg"
+import { logger } from "shared"
+import type { PromiseResult } from "shared-types"
+
+export interface Dictionary<T> {
```

Comment:
[Minor] We have a similar type in `shared-types` called `KeyValuePair<TKey, TValue>`.